Modules/_testcapimodule.c

```diff
@@ -3987,6 +3987,11 @@ test_structseq_newtype_doesnt_leak(PyObject *Py_UNUSED(self),
     assert(PyType_FastSubclass(structseq_type, Py_TPFLAGS_TUPLE_SUBCLASS));
     Py_DECREF(structseq_type);
 
+    descr.doc = NULL;
+    structseq_type = PyStructSequence_NewType(&descr);
+    assert(structseq_type != NULL);
+    Py_DECREF(structseq_type);
```

Comment:
If you want to add an unit test, I would be better to duplicate the HeapDocCType_spec test but with NULL tp_doc to test directly the PyType_FromSpec() function, since PyStructSequence_NewType() calls indirectly PyType_FromSpec().

See test_capi.test_heap_ctype_doc_and_text_signature().

Reply:
Make sense. Updated.